src/main/java/org/folio/service/file/upload/FileUploadServiceImpl.java

```diff
@@ -43,14 +56,34 @@ public FileUploadServiceImpl(@Autowired FileStorage fileStorage, @Autowired File
 
   @Override
   public Future<FileDefinition> saveFileChunk(FileDefinition fileDefinition, byte[] data, String tenantId) {
-    return fileStorage.saveFileDataAsync(data, fileDefinition);
+    if (data.length > 0) {
+      return fileStorage.saveFileDataAsync(data, fileDefinition)
+        .compose(ar -> updateFileDefinitionWithJobExecution(new JobExecution(), fileDefinition, tenantId));
+    }
+    return Future.succeededFuture(fileDefinition);
   }
 
   @Override
   public Future<FileDefinition> completeUploading(FileDefinition fileDefinition, String tenantId) {
-    JobExecution jobExecution = new JobExecution();
-    return jobExecutionService.save(jobExecution, tenantId).compose(savedJob ->
-      fileDefinitionService.update(fileDefinition.withStatus(COMPLETED).withJobExecutionId(savedJob.getId()), tenantId));
+    return fileDefinitionService.update(fileDefinition.withStatus(COMPLETED), tenantId);
+  }
+
+  @Override
+  public Future<FileDefinition> saveUUIDsByCQL(FileDefinition fileDefinition, String query, OkapiConnectionParams params) {
+    Optional<JsonObject> instancesUUIDs = inventoryClient.getInstancesBulkUUIDs(query, params);
+    List<String> ids = new ArrayList<>();
+    if (instancesUUIDs.isPresent()) {
+      JsonArray jsonIds = instancesUUIDs.get().getJsonArray("ids");
+      if (jsonIds.size() > 0) {
+        for (Object id : jsonIds) {
+          ids.add((String) id);
+        }
```

Comment:
just put it here
 new JobExecution().withProgress(new Progress().withTotal(String.valueOf(total)));